delphi/translators/for2py/rectify.py

```diff
@@ -997,39 +1027,46 @@ def handle_tag_statement(
-                else: 
+                    self.statements_to_reconstruct_before[
+                        'stmts-follow-label'].append(current)
+                else:
                     if (
-                        parent.tag == "body" 
-                        and parent.attrib['parent'] == "program"
+                            parent.tag == "body"
+                            and parent.attrib['parent'] == "program"
                     ):
-                        self.statements_to_reconstruct_before['stmts-follow-label'].append(current)
+                        self.statements_to_reconstruct_before[
+                            'stmts-follow-label'].append(current)
             elif self.label_after:
                 if self.collect_stmts_after_goto:
                     current.attrib['goto-remove'] = "true"
                     if "has-stop" not in current.attrib:
-                        self.statements_to_reconstruct_after['stmts-follow-goto'].append(current)
+                        self.statements_to_reconstruct_after[
+                            'stmts-follow-goto'].append(current)
                     else:
                         if (
-                            parent.tag == "body"
-                            and parent.attrib['parent'] == "program"
+                                parent.tag == "body"
+                                and parent.attrib['parent'] == "program"
                         ):
-                            self.statements_to_reconstruct_after['stmts-follow-goto'].append(current)
+                            self.statements_to_reconstruct_after[
+                                'stmts-follow-goto'].append(current)
 
                     if "goto-stmt" in current.attrib:
-                        self.stmts_after_goto['goto-stmts'].append(current.attrib['lbl'])
+                        self.stmts_after_goto['goto-stmts'].append(
+                            current.attrib['lbl'])
                     elif "target-label-statement" in current.attrib:
-                        self.stmts_after_goto['labels'].append(current.attrib['label'])
-                        
+                        self.stmts_after_goto['labels'].append(
+                            current.attrib['label'])
+
                 elif self.collect_stmts_after_label:
                     current.attrib['goto-remove'] = "true"
-                    self.statements_to_reconstruct_after['stmts-follow-label'].append(current)
-                    
+                    self.statements_to_reconstruct_after[
+                        'stmts-follow-label'].append(current)
+
                     if (
-                        (parent.tag == "body"
-                        and parent.attrib['parent'] == "program"
-                        and "has-stop" in current.attrib)
-                        or self.goto_under_loop
+                            (parent.tag == "body"
```

Comment:
parenthesize to make the operator grouping easy to understand.

Reply:
Relevant parenthesis has been added but since this code for derived types and written by @hlim1 , I would like him to confirm this change.

Reply:
I believe this is already been grouped. For example, `parent.tag == "body"` is 'A', `parent.attrib['parent'] == "program"` is B, and `"has-stop" in current.attrib is C`, and `self.goto_under_loop` is D. Then, with current parenthesis, it represents the grouping of (A and B and C) or D. However, I've given a separate individual lines for the parenthesis, so they can be more visually seen.

Reply:
@hlim1 Thanks, this is very helpful.  Can you also take a look at the boolean expression on lines 3493-3495 ?

Reply:
@skdebray I went through entire file just looking at the if-elif conditions and added grouping parenthesis for those seem to be ambiguously grouped or not having one, including line 3493-3495. However, if you find anything that I missed or you believe it can be done better, please let me know. Then, I will reflect it to the code immediately. Thank you!